python/packages/ai/teams/ai/models/openai_model.py

```diff
@@ -178,6 +178,9 @@ async def complete_prompt(
             messages.append(param)
 
         try:
+            extra_body = {}
```

Comment:
Hi @corinagum, as I understand some of the keys in `CompletionConfig` are used for prompt generation in `teams-ai`, and should probably not be included in the payload of the OpenAI API calls. So I don't know if we should just include any key. Your thought?

Reply:
@aacebo thoughts?

Reply:
I would, instead of whitelisting like this, just specifically not include the keys we use for our own purposes, so all keys other than those should be included. That way when someone for example brings their own custom model or OAI adds a new option we automatically support it without any extra code changes.

Reply:
I fully agree with the idea of generalizability but still see some implementation challenges. For example, OAI supports `extra_query`, `extra_body` and `extra_headers`  [request options](https://github.com/openai/openai-python/blob/main/README.md#undocumented-request-params) for undocumented params. So we wouldn't know how to put keys into the tree buckets without some kind of enumeration. This can be resolved if we modify `config.json` to add an `extra_body` key and move `data_sources` under it. But I don't know how feasible it is to make such a change.